### PR TITLE
Add a simple contract to support the memo feature.

### LINKED ISSUE #33: Address clippy warnings about using `panic!` without a string literal.
**Task description**

In some cases for the test infrastructure we use the `panic!` macro in `no_std` mode with a message that is not a string literal, but it is a string formatted using `format`.  This causes clippy to complain and might become an error in the future. We should try to fix this.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Add an example contract that can support a "memo" feature.

Also fixes #33 

## Changes

Add a new example smart contract.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: examples/memo/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+///
+/// And if both of these are valid it forwards the amount it received to the
+/// owner account.
+
+#[derive(Serialize, SchemaType)]
+/// The contract has no initialization parameters.
+struct InitParameter;
+
+/// Init function that creates a new contract.
+#[init(contract = "memo", parameter = "InitParameter", low_level)]
```
Comment (limemloh) [inline examples/memo/src/lib.rs:18]: You could save a few bytes by not adding the parameter to the schema, assuming you are embedding the schema.

File: examples/memo/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/// Receive a transaction with a message. This ensures that the message is 32
+/// bytes and that the sender of the message is an account.
+#[receive(contract = "memo", name = "receive", parameter = "ReceiveParameter", payable, low_level)]
+fn memo_receive<A: HasActions>(
+    ctx: &impl HasReceiveContext,
+    amount: Amount,
+    _state: &mut impl HasContractState,
+) -> ReceiveResult<A> {
+    matches!(ctx.sender(), Address::Account(..));
```
Comment (limemloh) [inline examples/memo/src/lib.rs:35]: I think you want to use `ensure!` around matches, otherwise this should not do anything

**Review by limemloh — CHANGES_REQUESTED** [review]

File: examples/memo/src/lib.rs (hunk 1 above)
Comment (abizjak) [reply examples/memo/src/lib.rs:18]: Yes, but I wanted to add it for symmetry.

File: examples/memo/src/lib.rs (hunk 2 above)
Comment (abizjak) [reply examples/memo/src/lib.rs:35]: Indeed, silly me.

**Review by limemloh — APPROVED** [review]

**Pull request merged**